### PR TITLE
Allow send to send reason – Resolves #198

### LINKED ISSUE #198: Can't pass reason to cfn custom resource response
the send function doesnt allow sending a reason

### PR DESCRIPTION
**Pull request by corrjo**
#198

Added reason to send module

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by corrjo** [comment]
Would love some review. Let me know if there is anything needs to be done. Thanks!

**Comment by jaymecd** [comment]
Very useful update, looking forward for the changes is publicly available.
It's important to expose failure reason back to CFN, as LogGroup might be deleted on failed creation, if it is a part of the template.

File: aws/services/CloudFormation/MacrosExamples/StackMetrics/lambda/cfnresponse.py [hunk 1]
```diff
@@ -11,14 +11,16 @@
 SUCCESS = "SUCCESS"
 FAILED = "FAILED"
 
-def send(event, context, responseStatus, responseData, physicalResourceId=None, noEcho=False):
+def send(event, context, responseStatus, responseData, reason=None, physicalResourceId=None, noEcho=False):
```
Comment (jaymecd) [inline aws/services/CloudFormation/MacrosExamples/StackMetrics/lambda/cfnresponse.py:14]: Suggest to keep existing signature by adding reason as last optional argument. 
That allows to keep existing implementations safe and w/o side-effects.

```suggestion
def send(event, context, responseStatus, responseData, physicalResourceId=None, noEcho=False, reason=None):
```

**Review by jaymecd — CHANGES_REQUESTED** [review]
pls check https://github.com/awslabs/aws-cloudformation-templates/pull/199/files#r520777025 suggestion

File: aws/services/CloudFormation/MacrosExamples/StackMetrics/lambda/cfnresponse.py (hunk 1 above)
Comment (corrjo) [reply aws/services/CloudFormation/MacrosExamples/StackMetrics/lambda/cfnresponse.py]: Great point!

**Review by jaymecd — APPROVED** [review]

**Comment by fraghag** [comment]
So this has been approved but never merged. Who has got the power to do that? There is also another PR that's adding the same thing (#382). So you can kill two PRs with one merge.

**Comment by gene1wood** [comment]
Also make sure that if this does get merged here in this template repo, that that change will also be reflected in the `cfnresponse` module present in the deployed AWS Lambda runtime environments.

**Comment by gene1wood** [comment]
#382 was merged. This PR, #198 can be closed

**Pull request closed without merging**